src/edu/stanford/nlp/sempre/FilterFormula.java

```diff
@@ -0,0 +1,64 @@
+package edu.stanford.nlp.sempre;
+
+import com.google.common.base.Function;
+import fig.basic.LispTree;
+
+import java.util.List;
+
+/**
+ * (filter x y): Find the set of all elements in x that function y evaluates to a non-empty set.   
```

Comment:
Give an example...in particular, so that people know the directionality of y (which should be consistent with SuperlativeFormula).